Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -247,18 +278,12 @@ extension TypeSyntax {
       return .implicitlyUnwrappedOptional(typeIdentifier.wrappedType.typeDescription)
 
     } else if let typeIdentifier = self.as(ArrayTypeSyntax.self) {
-      // An array of the form [Type] is synonymous with Array<Type>.
-      // Utilize a simple type rather than creating another enum case.
-      let elementDescription = typeIdentifier.elementType.typeDescription.asSource
-      return .simple(name: "Array<\(elementDescription)>")
+      return .array(element: typeIdentifier.elementType.typeDescription)
 
     } else if let typeIdentifier = self.as(DictionaryTypeSyntax.self) {
-      // An array of the form [KeyType: ValueType] is synonymous with Dictionary<KeyType, ValueType>.
-      // Utilize a simple type rather than creating another enum case.
-      let keyDescription = typeIdentifier.keyType.typeDescription
-      let valueDescription = typeIdentifier.valueType.typeDescription
-      return .simple(
-        name: "Dictionary<\(keyDescription.asSource), \(valueDescription.asSource)>")
+      return .dictionary(
+        key: typeIdentifier.keyType.typeDescription,
+        value: typeIdentifier.valueType.typeDescription)
```

Comment:
Another possible option here is to encode this type as: 
```suggestion
      return .simple(
        name: "Dictionary",
        generics: [typeIdentifier.keyType.typeDescription, typeIdentifier.valueType.typeDescription])
```
This would allow us to get rid of the `.array` case. We'd lose the context that we _know_ that we're dealing with a `Foundation.Array` type, but this way we'd ensure that `Array<Int>` and `[Int]` are treated the same way by the system. Thoughts?